index.html

```diff
@@ -35,58 +41,150 @@
+          <td>`null`
+          <td>A {{Promise}} used to wait for the socket to close or error. Corresponds to the {{TCPSocket/closed}} member.
+      </table>
+
+      <section>
+        <h3><dfn>constructor()</dfn> method</h3>
+
+        <aside class="example">
+          In order to communicate via TCP a socket connection must be requested first. The socket object constructor
+          allowes the site to specify the necessary parameters which control how data is transmitted and received.
+
+          <pre class="js">
+            const socket = new TCPSocket(/*remoteAddress=*/, /*remotePort=*/, /*options=*/);
+          </pre>
+
+          The developer should wait for the connection to be established before accessing the {{TCPSocket/connection}} attribute. 
+
+          <pre class="js">
+            const { readable, writable } = await socket.connection;
+          </pre>
+
+          Once {{TCPSocket/connection}} has resolved, the {{TCPSocketConnection/readable}} and {{TCPSocketConnection/writable}}
+          can be accessed to get the {{ReadableStream}} and {{WritableStream}} instances for receiving data from and sending data to the connected socket.
+        </aside>
+
+        The {{TCPSocket/constructor()}} steps are:
+
+        <ol>
+          <li>If [=this=]'s [=relevant global object=]'s [=associated Document=] is
+        not [=allowed to use=] the [=policy-controlled feature=] named "[=policy-controlled feature/direct-sockets=]", throw a
+        "{{NotAllowedError}}" {{DOMException}} and return `null`.
+          <li>If only one of |options|["{{TCPSocketOptions/localAddress}}"] or |options|["{{TCPSocketOptions/localPort}}"] is specified, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/keepAlive}}"] is specified and |options|["{{TCPSocketOptions/keepAliveDelay}}"] is not, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/keepAlive}}"] is not specified and |options|["{{TCPSocketOptions/keepAliveDelay}}"] is, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/keepAliveDelay}}"] is less than one second, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/sendBufferSize}}"] is equal to 0, throw a {{TypeError}}  and return `null`.
+          <li>If |options|["{{TCPSocketOptions/receiveBufferSize}}"] is equal to 0, throw a {{TypeError}}  and return `null`.
+          <li>Perform the following steps [=in parallel=].
+            <ol>
+              <li>
+                Invoke the operating system to open the TCP socket using the given |remoteAddress| and |remotePort| and the 
+                connection parameters (or their defaults) specified in |options|.
+              <li>
+                If this fails for any reason:
+                <ol>
+                  <li>[=Reject=] the {{TCPSocket/[[connectionPromise]]}} with a "{{NetworkError}}" {{DOMException}}.
```

Comment:
Because these steps are running in parallel you need the "queue a task on..." incantation to make resolving or rejecting the promise happen back on the main thread.